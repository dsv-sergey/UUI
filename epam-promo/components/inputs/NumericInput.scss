@use  '../../assets/styles/index.scss' as *;
@import '../../assets/styles/colorvars/inputs/numericInput-colorvars.scss';

.root {
    @include numeric-input-color();

    &:global(.uui-input-box) {
        padding-right: 0;
        min-width: 40px;

        :global(.uui-numeric-input-up-button) {
            transform: rotate(180deg);

            &:hover {
                svg {
                    fill: var(--icon-color-hovered);
                }
            }
        }

        :global(.uui-numeric-input-down-button) {
            &:hover {
                svg {
                    fill: var(--icon-color-hovered);
                }
            }
        }

        :global(.uui-icon) svg {
            margin-top: -2px;
        }

        &.size-24 {
            :global(.uui-numeric-input-button-group) {
                > * {
                    height: 12px;
                }
            }
        }

        &.size-30 {
            :global(.uui-numeric-input-button-group) {
                > * {
                    height: 15px;
                }
            }
        }

        &.size-36 {
            :global(.uui-numeric-input-button-group) {
                > * {
                    height: 17px;
                }
            }
        }

        &.size-42 {
            :global(.uui-numeric-input-button-group) {
                > * {
                    height: 21px;
                }
            }
        }

        &.size-48 {
            :global(.uui-numeric-input-button-group) {
                > * {
                    height: 23px;
                }
            }
        }
    }

    &:global(.uui-disabled) {
        :global(.uui-numeric-input-button-group) {
            > *:hover, > *:focus {
                cursor: default;
                background-color: transparent;
            }
        }
    }
}
