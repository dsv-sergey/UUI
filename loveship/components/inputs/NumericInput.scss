@use '../../assets/styles/scss/index'  as *;
@import 'TextInput';

.root {
    &:global(.uui-focus) {
        @include control-focus-effect();
        border-color: $sky;

        &.mode-cell {
            :global(.uui-numeric-input-button-group) {
                & > :first-child:hover, & > :first-child:focus {
                    box-shadow: inset 1px -1px 0 1px rgba(48, 182, 221, 0.4);
                }

                & > :last-child:hover, & > :last-child:focus {
                    box-shadow: inset -1px -1px 0 1px rgba(48, 182, 221, 0.4);
                }
            }
        }
    }

    &:global(.uui-input-box) {
        padding-right: 0;
        min-width: 40px;

        :global(.uui-numeric-input-up-button) {
            transform: rotate(180deg);
        }

        :global(.uui-numeric-input-button-group) {
            & > *:hover, & > *:focus {
                outline: none;
                background-color: $night50;
            }
        }

        :global(.uui-icon) svg {
            margin-top: -2px;
        }

        &.size-24 {
            :global(.uui-numeric-input-button-group) {
                & > * {
                    height: 11px;
                }
            }
        }

        &.size-30 {
            :global(.uui-numeric-input-button-group) {
                & > * {
                    height: 14px;
                }
            }
        }

        &.size-36 {
            :global(.uui-numeric-input-button-group) {
                & > * {
                    height: 17px;
                }
            }
        }

        &.size-48 {
            :global(.uui-numeric-input-button-group) {
                & > * {
                    height: 23px;
                }
            }
        }
    }

    &:global(.uui-disabled) {
        :global(.uui-numeric-input-button-group) {
            & > *:hover, & > *:focus {
                cursor: default;
                background-color: transparent;
            }
        }
    }

    &:global(.uui-readonly) {
        @include readonly-input();
    }
}
