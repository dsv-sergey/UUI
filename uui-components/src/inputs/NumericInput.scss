.container {
    input::-webkit-outer-spin-button,
    input::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
    }

    &:global(.uui-input-box) {
        position: relative;
        display: flex;
        outline: none;

        :global(.uui-input) {
            -moz-appearance: textfield;
            flex: 1 1 auto;
            min-width: 0;
            border: 0;
            outline: none;
            background: none;

            &::-ms-clear {
                display: none;
            }

            &:invalid {
                box-shadow: none;
            }
        }

        :global(.uui-numeric-input-up-button), :global(.uui-numeric-input-down-button) {
            &:hover {
                cursor: pointer;
            }
        }
    }
}

.hidden {
    visibility: hidden;
}