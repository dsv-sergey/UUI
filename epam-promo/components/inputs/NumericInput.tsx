import * as types from '../types';
import * as css from './NumericInput.scss';
import * as textInputCss from './TextInput.scss';
import { withMods } from '@epam/uui';
import { NumericInput as uuiNumericInput, NumericInputProps } from '@epam/uui-components';
import { systemIcons } from '../../icons/icons';

const defaultSize = '36';

export interface NumericInputMods extends types.SizeMod {
}

export function applyNumericInputMods(mods: NumericInputMods) {
    return [
        textInputCss.root,
        css.root,
        css['size-' + (mods.size || defaultSize)],
        textInputCss['size-' + (mods.size || defaultSize)],
    ];
}

export const NumericInput = withMods<NumericInputProps, NumericInputMods>(
    uuiNumericInput,
    applyNumericInputMods,
    (props) => ({
        upIcon: systemIcons[props.size || defaultSize].foldingArrow,
        downIcon: systemIcons[props.size || defaultSize].foldingArrow,
    }),
);