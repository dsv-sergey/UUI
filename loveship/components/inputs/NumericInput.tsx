import * as types from '../types';
import * as css from './NumericInput.scss';
import * as colorStyle from '../../assets/styles/scss/loveship-color-vars.scss';
import { withMods } from '@epam/uui';
import { NumericInput as uuiNumericInput, NumericInputProps } from '@epam/uui-components';
import { TextSettings, getTextClasses } from '../../helpers/textLayout';
import { systemIcons } from '../icons/icons';

const defaultSize = '36';

export interface NumericInputMods extends types.EditMode, types.SizeMod, TextSettings {
}

export function applyNumericInputMods(mods: NumericInputMods) {
    return [
        colorStyle.colorSky,
        css.root,
        css['size-' + (mods.size || defaultSize)],
        css['mode-' + (mods.mode || 'form')],
    ];
}

export const NumericInput = withMods<NumericInputProps, NumericInputMods>(
    uuiNumericInput,
    applyNumericInputMods,
    (props) => ({
        upIcon: systemIcons[props.size || defaultSize].foldingArrow,
        downIcon: systemIcons[props.size || defaultSize].foldingArrow,
        inputCx: getTextClasses({
            size: props.size || defaultSize,
            lineHeight: props.lineHeight,
            fontSize: props.fontSize,
        }, true),
    }),
);