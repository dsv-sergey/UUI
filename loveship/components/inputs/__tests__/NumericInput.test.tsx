import React from "react";
import renderer from "react-test-renderer";
import {NumericInput} from "../NumericInput";

describe("NumericInput", () => {
    const onChange = jest.fn();

    it("should be rendered correctly", () => {
        const tree = renderer
            .create(<NumericInput
                min={ 1 }
                max={ 10 }
                value={ 5 }
                onValueChange={ onChange }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be rendered correctly with new props", () => {
        const tree = renderer
            .create(<NumericInput
                min={ 1 }
                max={ 10 }
                value={ 5 }
                onValueChange={ onChange }
                size="24"
                mode="cell"
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});