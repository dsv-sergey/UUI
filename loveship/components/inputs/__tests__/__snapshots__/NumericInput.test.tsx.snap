// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput should be rendered correctly 1`] = `
<div
  className="container uui-input-box colorSky root size-36 mode-form"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input line-height-24 font-size-14 v-padding-5"
    max={10}
    min={1}
    onChange={[Function]}
    placeholder="0"
    step={1}
    type="number"
    value="5"
  />
  <div
    className="uui-numeric-input-button-group"
  >
    <div
      className="container uui-icon uui-enabled -clickable uui-numeric-input-up-button"
      onClick={[Function]}
    >
      <svg
        className=""
        height={24}
        viewBox="0 0 12 24"
        width={12}
      >
        <use
          xlinkHref="#folding-arrow-18.svg"
        />
      </svg>
    </div>
    <div
      className="container uui-icon uui-enabled -clickable uui-numeric-input-down-button"
      onClick={[Function]}
    >
      <svg
        className=""
        height={24}
        viewBox="0 0 12 24"
        width={12}
      >
        <use
          xlinkHref="#folding-arrow-18.svg"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`NumericInput should be rendered correctly with new props 1`] = `
<div
  className="container uui-input-box colorSky root size-24 mode-cell"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  tabIndex={-1}
>
  <input
    className="uui-input line-height-18 font-size-12 v-padding-2"
    max={10}
    min={1}
    onChange={[Function]}
    placeholder="0"
    step={1}
    type="number"
    value="5"
  />
  <div
    className="uui-numeric-input-button-group"
  >
    <div
      className="container uui-icon uui-enabled -clickable uui-numeric-input-up-button"
      onClick={[Function]}
    >
      <svg
        className=""
        height={24}
        viewBox="0 0 12 24"
        width={12}
      >
        <use
          xlinkHref="#folding-arrow-12.svg"
        />
      </svg>
    </div>
    <div
      className="container uui-icon uui-enabled -clickable uui-numeric-input-down-button"
      onClick={[Function]}
    >
      <svg
        className=""
        height={24}
        viewBox="0 0 12 24"
        width={12}
      >
        <use
          xlinkHref="#folding-arrow-12.svg"
        />
      </svg>
    </div>
  </div>
</div>
`;
